python/requirements.txt

```diff
@@ -0,0 +1,3 @@
+Sphinx>=4.0.2
+pyarrow>=4.0.0
```

Comment:
There was some discussion in place with Nic, and there is a similar issue for the doc itself ( https://issues.apache.org/jira/browse/ARROW-13411 ). The two problems are probably going to share the same solution (at least the principles) so for the moment we didn't move into any specific direction for the cookbook. It will always compile against the latest released arrow version, which is usually what we want (or the CI can explicitly preinstall the master branch so that it can get checked against that one)